src/fsharp/service/ServiceParamInfoLocations.fsi

```diff
@@ -33,8 +36,13 @@ type public FSharpNoteworthyParamInfoLocations =
     member IsThereACloseParen : bool   
 
     /// Either empty or a name if an actual named parameter; f(0,a=4,?b=None) would be [|None; Some "a"; Some "b"|]
-    member NamedParamNames : string option []  
+    member NamedParamNames : string option []
+
+    /// Array of locations for each argument, and a flag if that argument is named
+    member ArgLocations: TupledArgumentLocation []
 
     /// Find the information about parameter info locations at a particular source location
     static member Find : pos * ParsedInput -> FSharpNoteworthyParamInfoLocations option
 
+module internal FunctionApplicationArgumentLocationsImpl =
```

Comment:
In the terminology of the compiler the terminology `FunctionApplication` sounds wrong as that would not apply to member applications.  Also it's not clear if it applies to `f x` where `f` is a first class function value without argument name data.

It looks to me like it will apply to `SynExpr.App` nodes, except `a + b`.  

1. I believe this should be extended to curried applications.  They shouldn't be treated differently by this sort of rutine

2. Be aware of prefix operators `-x` and `~~~x`

3. I think use terminology `App`  to match `SynExprApp`, so `SynExprAppLocationsImpl` and `GetAllCurriedArgsAtPosition`

Reply:
>  except a + b.

This is actually accounted for, since you'd want to track the `x` in something like `string x + y`. the `x + string y` case is already accounted for though.

Will need to check prefix operators, thanks.